### ISSUE TITLE
Blurred results when drawing geometries

### ISSUE THREAD

**Issue by vladimir-stoyanov** [microsoft/Win2D#879]
Hello,

I will start by saying that this API is very useful to us here at Telerik. We have been using it to draw text, images and shapes coming from a PDF document.

The issue we have is that the output produced when drawing a geometry is blurred. This is especially noticeable when the drawn geometries are scaled (for example with a RenderTransform), however the blurriness is visible even without scaling. The approach we have adopted is drawing a CanvasRenderTarget inside the Draw event of a CanvasControl. 

I am attaching a small sample project([Win2DTestProject.zip](https://github.com/microsoft/Win2D/files/8813737/Win2DTestProject.zip)), where you can observe a difference in the quality of a drawn line when it is drawn through a CanvasRenderTarget or directly inside the Draw method of a CanvasControl. 

Can you check out the shared sample project and let me know, if it is possible to reduce the blurriness of the drawn geometry when using a CanvasRenderTarget? 

Looking forward to your reply.

**Comment by vladimir-stoyanov** [comment]
Hello, 

Thank you for the tips. Indeed changing the DPI or avoiding it entirely helped out. 

That said, I would appreciate any other tips to improve the crispness of the drawn geometry. Comparing the results between drawing the same thing in WinUI and WPF, it seems that the WinUI version is more blurry. I know it's a different technology, however I might be missing something that can improve the quality. 

Additionally, can you share whether the quality of the drawn geometry can be improved when scaling it natively within the framework (by using a ScaleTransform on the CanvasControl)?

Thank you again for your help :)

**Comment by dimohy** [comment]
Hello.

I have the same problem and there is no solution.

I'm using 96 DPI fixedly and I also deliver it to CanvasRenderTarget as 96 DPI, so theoretically, I think there should be no change in quality.

**Comment by kalin-todorov** [comment]
Hi @getrou,

> 2. Scaling up your CanvasControl using CanvasControl.RenderTransform is further blurring your images. I think from your question you understand that and aren't concerned about this part, but if you are let me know and I can help you reduce that as well!
> 
> TLDR - Pass a higher DPI into your CanvasRenderTarget constructor, and let me know if you need more help!

Do you have any updates about the second point? Is it possible to look better when scaled using RenderTransform?

Thanks,
Kalin

**Comment by getrou** [comment]
Aha, found it. Just took me a minute to find.

Rather than using a RenderTransform to scale your CanvasControl, use the following:
args.DrawingSession.Transform = Matrix3x2.CreateScale(2.5f, 2.5f);

This will give you crisp SVG drawing, rather than scaling up the rasterization of the drawn geometry.
This won't fix the scaling of your CanvasRenderTarget. You'll still need to increase the size of that intermediate render target if you want to scale it up. But your regular geometry calls will be nice and crisp at >1.0 scale.

**Comment by vladimir-stoyanov** [comment]
Hello @getrou,

Indeed I have tried using the Transform property and the visualization is very crisp. However in our use-case scenario we would need to do some more modifications to adjust the clipping and the positioning of the drawn geometries, if we go with that.

That is why I tried the DpiScale of the CanvasControl and it also provides a sufficient improvement for now without requiring any further changes. With this in mind we have adopted this approach for the time being. 

Thank you very much for your suggestions :)
